### ISSUE TITLE
pandas DataFrameGroupby.agg (still) has incomplete signature

### ISSUE THREAD

**Issue by tuchandra** [microsoft/pylance-release#2180]
I originally put this in a [comment](https://github.com/microsoft/pylance-release/issues/2080#issuecomment-994108294) on #2080, but I figured I'd make a standalone issue.

<h3> Environment data </h3>

-   Language Server version: 2021.12.2-pre.1
-   OS and version: darwin x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.9.9, via pyenv
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: basic

<h3> Expected behaviour </h3>
That I can pass two `NamedAgg`s to the `agg` method of `DataFrameGroupBy`.

<h3> Actual behaviour </h3>
I get an error -

```
No overloads for "agg" match the provided arguments
  Argument types: (NamedAgg, NamedAgg)PylancereportGeneralTypeIssues
```

<h3> Logs </h3>

Python Language Server Log
```
Background analysis message: setFileOpened
Background analysis message: markFilesDirty
Background analysis message: analyze
[BG(1)] analyzing: Untitled-1 ...
[BG(1)]   parsing: Untitled-1 (5ms)
[BG(1)]   binding: Untitled-1 (2ms)
[BG(1)]   checking: Untitled-1 ...
[BG(1)]     parsing: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/__init__.pyi [fs read 2ms] (15ms)
[BG(1)]     binding: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/__init__.pyi (1ms)
[BG(1)]     parsing: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/api.pyi [fs read 1ms] (4ms)
[BG(1)]     binding: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/api.pyi (0ms)
[BG(1)]     parsing: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/frame.pyi [fs read 1ms] (75ms)
[BG(1)]     binding: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/frame.pyi (14ms)
[BG(1)]     parsing: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/generic.pyi [fs read 1ms] (10ms)
[BG(1)]     binding: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/generic.pyi (3ms)
[BG(1)]     parsing: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/base.pyi [fs read 1ms] (3ms)
[BG(1)]     binding: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/base.pyi (1ms)
[BG(1)]     parsing: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/accessor.pyi [fs read 1ms] (1ms)
[BG(1)]     binding: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/accessor.pyi (0ms)
[BG(1)]     parsing: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/indexing.pyi [fs read 0ms] (1ms)
[BG(1)]     binding: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/indexing.pyi (0ms)
[BG(1)]     parsing: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/_typing.pyi [fs read 1ms] (4ms)
[BG(1)]     binding: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/_typing.pyi (1ms)
[BG(1)]     parsing: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/indexes/api.pyi [fs read 1ms] (2ms)
[BG(1)]     binding: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/indexes/api.pyi (0ms)
[BG(1)]     parsing: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/indexes/base.pyi [fs read 1ms] (4ms)
[BG(1)]     binding: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/indexes/base.pyi (2ms)
[BG(1)]     parsing: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/series.pyi [fs read 1ms] (17ms)
[BG(1)]     binding: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/series.pyi (10ms)
[BG(1)]     parsing: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/typeshed-fallback/stdlib/datetime.pyi [fs read 1ms] (6ms)
[BG(1)]     binding: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/typeshed-fallback/stdlib/datetime.pyi (2ms)
[BG(1)]     parsing: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/groupby/__init__.pyi [fs read 0ms] (1ms)
[BG(1)]     binding: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/groupby/__init__.pyi (0ms)
[BG(1)]     parsing: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/groupby/generic.pyi [fs read 1ms] (10ms)
[BG(1)]     binding: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/groupby/generic.pyi (4ms)
[BG(1)]     parsing: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/groupby/groupby.pyi [fs read 1ms] (2ms)
[BG(1)]     binding: /Users/tushar.chandra/.vscode/extensions/ms-python.vscode-pylance-2021.12.3-pre.1/dist/bundled/stubs/pandas/core/groupby/groupby.pyi (3ms)
[BG(1)]   checking: Untitled-1 (248ms)
[BG(1)] analyzing: Untitled-1 (256ms)
Background analysis message: getSemanticTokens delta
[BG(1)] getSemanticTokens delta previousResultId:1639680019523 at Untitled-1 (2ms)
Background analysis message: resumeAnalysis
```

<h3> Code Snippet / Additional information </h3>

```
import pandas as pd

df = pd.DataFrame()
group = df.groupby("x")
reveal_type(group)  # DataFrameGroupBy
agged = group.agg(x=pd.NamedAgg("a", "b"), y=pd.NamedAgg("c", "d"))
```

![image](https://user-images.githubusercontent.com/11623486/146090540-6329d495-b0e2-4808-94d3-5a91fc8ae7c4.png)

**Comment by bschnurr** [comment]
Thanks for the separate issue. the type stub for `agg` needs to be updated and I also think my previous update to `NamedAgg` also needs to be updated.

**Comment by debonte** [comment]
This issue has been fixed in version 2022.2.0, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202220-2-February-2022)

**Comment by hugepower** [comment]
<img width="816" alt="image" src="https://user-images.githubusercontent.com/46076022/152626478-3bb789de-2429-45ed-9b0e-6a9f62a4d6f6.png"> 

It didn't work.
